packages/xforms-engine/src/parse/body/control/InputControlDefinition.ts

```diff
@@ -11,10 +11,16 @@ export class InputControlDefinition extends ControlDefinition<'input'> {
 
 	readonly type = 'input';
 	readonly appearances: InputAppearanceDefinition;
+	readonly rows: number | null;
+	readonly accuracyThreshold: number | null;
+	readonly unacceptableAccuracyThreshold: number | null;
 
 	constructor(form: XFormDefinition, parent: BodyElementParentContext, element: Element) {
 		super(form, parent, element);
 
 		this.appearances = inputAppearanceParser.parseFrom(element, 'appearance');
+		this.rows = Number(element.getAttribute('rows')); // ToDo: Make parser function, validate NaN
+		this.accuracyThreshold = Number(element.getAttribute('accuracyThreshold')); // ToDo: Make parser function, validate NaN
+		this.unacceptableAccuracyThreshold = Number(element.getAttribute('unacceptableAccuracyThreshold')); // ToDo: Make parser function, validate NaN
```

Comment:
Not suggesting a change this early on. Just adding a minor "yes and" on these TODO comments for additional context. There are two gaps here, and they apply in a few other places too:

- `accuracyThreshold` and `unacceptableAccuracyThreshold` are specified as decimals. We'll want to make sure that whatever parsing we do accounts for that (including potential float rounding errors).
- `rows` doesn't specify its type explicitly, but I think it's safe to assume it should be an integer. Same thing about parsing the type, different type to account for.

I wish we'd been tracking these cases sooner, because like I said there are others!

Reply:
Sorry, I meant to come back to this before. I meant in the project in general. My intent was not to change anything in how the attributes are parsed, just to add context to the TODO comments as they were already written.